src/data_chain/chain.rs

```diff
@@ -185,3 +190,132 @@ impl Chain {
+    use full_info::FullInfo;
+    use maidsafe_utilities::SeededRng;
+    use routing_table::Prefix;
+    use rust_sodium;
+    use rust_sodium::crypto::sign;
+    use tiny_keccak::sha3_256;
+    use xor_name::XorName;
+
+    #[test]
+    fn is_valid_pair() {
+        let mut rng = SeededRng::thread_rng();
+        unwrap!(rust_sodium::init_with_rng(&mut rng));
+
+        let group_size = 8;
+        let chain = Chain::from_blocks(Vec::new(), group_size);
+
+        let full_info = FullInfo::node_new(0);
+
+        let (pub_key_0, _) = sign::gen_keypair();
+        let name = XorName(sha3_256(&pub_key_0.0));
+        let prefix = Prefix::new(1, name, 0);
+
+        let node_state = NodeState {
+            state: State::ElderLive,
+            public_key: pub_key_0,
+            age: 5,
+            section: prefix,
+        };
+
+        // Check valid pairs of blocks.
+        let valid_pairs = [
+            (State::ElderLive, State::ElderOffline),
+            (
+                State::ElderLive,
+                State::ElderSplitTo { to: prefix.pushed(true) },
+            ),
+            (State::ElderLive, State::ElderDemoted),
+            (State::ElderLive, State::ElderRelocated),
+            (
+                State::ElderSplitTo { to: prefix.pushed(true) },
+                State::ElderLive,
+            ),
+        ];
+        for &(first, second) in &valid_pairs {
+            let first = make_block(
+                &full_info,
+                NodeState {
+                    state: first,
+                    ..node_state
+                },
+            );
+            let second = make_block(
+                &full_info,
+                NodeState {
+                    state: second,
+                    ..node_state
+                },
+            );
+
+            assert!(chain.is_valid_pair(&first, &second));
+        }
+
+        // Check invalid pairs of blocks.
+        let invalid_pairs = [
+            (State::ElderOffline, State::ElderLive),
+            (
+                State::ElderOffline,
+                State::ElderSplitTo { to: prefix.pushed(true) },
+            ),
+            (State::ElderOffline, State::ElderDemoted),
+            (State::ElderOffline, State::ElderRelocated),
+            (State::ElderDemoted, State::ElderLive),
+            (State::ElderRelocated, State::ElderLive),
+        ];
+        for &(first, second) in &invalid_pairs {
+            let first = make_block(
+                &full_info,
+                NodeState {
+                    state: first,
+                    ..node_state
+                },
+            );
+            let second = make_block(
+                &full_info,
+                NodeState {
+                    state: second,
+                    ..node_state
+                },
+            );
+
+            assert!(!chain.is_valid_pair(&first, &second));
+        }
+
+        // Pair is only valid if both blocks refer to the same node.
```

Comment:
I think it's the opposite to this, but I don't think the `is_valid_pair()` method even needs to check for inequality - it's only interested in the variant types.